added display none style on hide class in image_upload_widget.css

mjmanas54 opened the pull request:
PR fixes the issue of rendering of "upload new profile picture" on hovering on user avatar even if user does not have permission to change user avatar.
Fixes: #23844 

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
Before:
![Screenshot from 2022-12-13 13-06-01](https://user-images.githubusercontent.com/101850957/207774733-aee2e51f-3e5f-446c-9494-81ef5d834374.png)
After:
![Screenshot from 2022-12-15 10-17-33](https://user-images.githubusercontent.com/101850957/207774913-e2fef783-30d3-4240-8630-af1bdfa553a7.png)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

mjmanas54 commented:
My PR is ready. @alya , would you be up for reviewing this?

alya commented:
Looks good to me in manual testing! @sahil839 could you please review this one?

sahil839 commented:
Code looks good to me. Commit message should be updated to follow the [guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages).

sahil839 commented:
Also, we should fix the live-update of the widget when the setting is changed. I think we should do the following changes -
- We should update `update_avatar_change_display`, to add/remove the `hide` class for `image_upload_button` element instead of calling show/hide function. The `show` function adds `display: block` class while we use `display: flex` class for the `image_upload_button`, so the `Upload new profile picture` text is misplaced till page is reloaded. Also, make sense to remove `hide` class when user is allowed to change avatar.
- The `Avatar changes are disabled in this organization` text is also not live-updated, it is shown/hidden only after page is reloaded. We should add the code to hide/show that particular `div` element in `update_avatar_change_display` function.

The above mentioned changes can be done in separate commits.

mjmanas54 commented:
@sahil839 As you mentioned in first point
I have updated `settings_accout.js`
changed `$("#user-avatar-upload-widget .image_upload_button").hide();`
to `$("#user-avatar-upload-widget .image_upload_button").addClass("hide");`
and `$("#user-avatar-upload-widget .image_upload_button").show();`
to `$("#user-avatar-upload-widget .image_upload_button").addClass("hide");`
and I am getting following error in node_tests

![Screenshot from 2023-01-03 23-11-13](https://user-images.githubusercontent.com/101850957/210412780-4a067d3f-d3b8-4a58-9362-d8779a533098.png)

Please help me to sort it out.



> Also, we should fix the live-update of the widget when the setting is changed. I think we should do the following changes -
> 
> * We should update `update_avatar_change_display`, to add/remove the `hide` class for `image_upload_button` element instead of calling show/hide function. The `show` function adds `display: block` class while we use `display: flex` class for the `image_upload_button`, so the `Upload new profile picture` text is misplaced till page is reloaded. Also, make sense to remove `hide` class when user is allowed to change avatar.
> * The `Avatar changes are disabled in this organization` text is also not live-updated, it is shown/hidden only after page is reloaded. We should add the code to hide/show that particular `div` element in `update_avatar_change_display` function.
> 
> The above mentioned changes can be done in separate commits.

sahil839 commented:
`.show` call should be replaced by `removeClass("hide")`

mjmanas54 commented:
Thanks @laurynmm .
